Address some issues with the Sockets tests.

pgavlin opened the pull request:
- Mark Disconnect() and SendPackets() tests as Windows-specific
- Mark a few tests that cover dual-stack semantics with a dummy issue number while the differences are investigated
- Fix port numbers on a multitude of tests to avoid conflicts at runtime
- Make the perf tests runnable on non-Windows platforms.

pgavlin committed:
Fix some issues with the Sockets tests.

- Mark Disconnect() and SendPackets() tests as Windows-specific
- Mark a few tests that cover dual-stack semantics with a dummy issue number
  while the differences are investigated
- Fix port numbers on a multitude of tests to avoid conflicts at runtime

pgavlin committed:
Remove Windows-specific code from Socket perf tests.

pgavlin commented:
@CIPop I'll be updating this PR to address the feedback you left on the original PR that included these changes. I'll ping you again once I've done so.

stephentoub commented:
I still question the manual assignment of a port number to each test.  Someone adding a new test is going to need to go and find the next available number, and it seems like there's a really good chance of accidentally using an already used number.  Even if using a a counter to pick the next port number has the same chance of conflicting with other things in the system, it still seems less at risk for these kinds of errors.

But it's up to you guys.  Other than that, LGTM.

pgavlin commented:
> I still question the manual assignment of a port number to each test.

As per the previous CR, I think the long-term plan is to move all of the tests to use ephemeral ports that are automatically assigned by the OS.

pgavlin committed:
Address CR feedback.

- Remove some extra using blocks and dispose calls to restore original
  test behavior
- Add tests for Disconnect{,Async} on non-Windows platforms
- Restore usage of memcmp for Windows performance tests

pgavlin commented:
@CIPop I believe I've addressed all comments from #19 with the latest commit in this PR.

pgavlin commented:
@davidsh can you take a look at these changes as @CIPop is on vacation?

davidsh commented:
@pgavlin I need to abstain from this PR.  I have not been involved in this particular effort and don't realy have context to give it justice reviewing it. Given the approaching deadline, I'm fine with pushing this if you think it is ready for the next step.  When @CIPop is back on Wednesday, he can give additional feedback.  You can incorporate that feedback later into future PR's..

pgavlin commented:
SGTM. Thanks!

CIPop commented:
LGTM

@pgavlin Thanks for splitting the memcmp part!

Regarding the port part, if there are still issues you could disable parallel execution:
See @davidsh code in `System.Net.Http.WinHttpHandler`: (`XunitTestAssemblyAtrributes.cs`):
`[assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly)]`

This doesn't address running various OS tests on the same box at the same time.
